modules/dnn/src/onnx/onnx_importer.cpp

```diff
@@ -808,6 +826,19 @@ Net readNetFromONNX(const String& onnxFile)
     return net;
 }
 
+Net readNetFromONNX(const char* buffer, size_t sizeBuffer)
+{
+    ONNXImporter onnxImporter(buffer, sizeBuffer);
+    Net net;
+    onnxImporter.populateNet(net);
+    return net;
+}
+
+Net readNetFromONNX(const std::vector<uchar>& buffer)
+{
+    return readNetFromONNX(reinterpret_cast<const char*>(buffer.data()), buffer.size());
```

Comment:
`std::vector::data` is a C++11 feature. @alalek, Do you have an idea why the tests are passed?

Reply:
GCC uses GNU extensions by default (`-std=gnu++98`).
MSVS2015+ STL has `.data()` method too.

ANSI C++98 compilation mode is not tested at this moment (and may not work, at least in protobuf codes).

---

We already [using this method in OpenCV](https://github.com/opencv/opencv/blob/3.4.6/modules/core/src/matrix.cpp#L377-L380).

Reply:
Ok, I see, thank you! So I think we can merge this PR if you have no extra comments.